src/script/main.ts

```diff
@@ -1,10 +1,26 @@
+import { stat } from "fs";
+import { isNullOrUndefined } from "util";
+
 export class Main {
     public static viPath;
+    public static viModel;
 
     public static LoadFile(filePath: string) {
-        const currentState = history.state;
+        const currentState = top.history.state;
         const search = "?q=" + filePath;
         if (currentState !== null) {
+            /* As few file models(like LVClass/ PolyVi) donot have # values.
+            So, the #value for them in session storage is set as empty string.
+            Otherwise, it gets automatically assigned to undefined.*/
+            if (isNullOrUndefined(currentState.filePath.split("#")[1])) {
```

Comment:
This can be simplified to:
>const hash = isNullOrUndefined(currentState.filePath.split("#")[1]) ? "" : currentState.filePath.split("#")[1]
const stateObj = {hashValue: currentState.filePath.split("#")[1],
                                  sourceModel: currentState.parsedContent};
                sessionStorage.setItem(currentState.filePath.split("#")[0], JSON.stringify(stateObj));

Correct me if I'm wrong, but that's the only difference in the whole code inside the if and else cases, right? If so, then you can just pull out that one thing and avoid repeating the common code.